File: transaction-bff/src/main/java/br/com/felipe/gadelha/coffeandit/transactionbff/domain/entity/DailyLimit.java
package br.com.felipe.gadelha.coffeandit.transactionbff.domain.entity;

import java.math.BigDecimal;
import java.time.LocalDate;

public class DailyLimit {

    private final Long id;
    private final String agency;
    private final String account;
    private final LocalDate date;
    private final BigDecimal value;

    @Deprecated
    private DailyLimit(Long id, String agency, String account, LocalDate date, BigDecimal value) {
        this.id = id;
        this.agency = agency;
        this.account = account;
        this.date = date;
        this.value = value;
    }
    private DailyLimit(Builder builder) {
        this.id = builder.id;
        this.agency = builder.agency;
        this.account = builder.account;
        this.date = builder.date;
        this.value = builder.value;
    }
    public static Builder builder() { return new Builder(); }
    public static class Builder {
        private Long id;
        private String agency;
        private String account;
        private LocalDate date;
        private BigDecimal value;

        public Builder id(Long id) {
            this.id = id;
            return this;
        }
        public Builder agency(String agency) {
            this.agency = agency;
            return this;
        }
        public Builder account(String account) {
            this.account = account;
            return this;
        }
        public Builder date(LocalDate date) {
            this.date = date;
            return this;
        }
        public Builder value(BigDecimal value) {
            this.value = value;
            return this;
        }
        public DailyLimit build() { return new DailyLimit(this); }
    }
    public Long getId() { return id; }
    public String getAgency() { return agency; }
    public String getAccount() { return account; }
    public LocalDate getDate() { return date; }
    public BigDecimal getValue() { return value; }
}

File: limit-ms/src/main/kotlin/br/com/felipe/gadelha/coffeandit/limitms/domain/entity/DailyLimit.kt
package br.com.felipe.gadelha.coffeandit.limitms.domain.entity

import java.math.BigDecimal
import java.time.LocalDate
import java.util.Objects
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "daily_limits")
data class DailyLimit(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    val agency: String,
    val account: String,
    val date: LocalDate,
    val value: BigDecimal
) {
    fun update(updated: DailyLimit) = DailyLimit(
        id = if (Objects.isNull(id)) this.id else updated.id,
        agency = updated.agency,
        account = updated.account,
        date = updated.date,
        value = updated.value
    )
}
